tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/AdoNetDefinitions.cs

```diff
@@ -3,56 +3,86 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet;
+using Datadog.Trace.Configuration;
 using static Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.AdoNetClientInstrumentMethodAttribute;
 using static Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.AdoNetConstants;
 
-/********************************************************************************
- * Task<int> .ExecuteNonQueryAsync(CancellationToken)
- ********************************************************************************/
+#pragma warning disable SA1118 // parameter spans multiple lines
+[assembly: AdoNetClientInstrumentMethod(
+    AssemblyName = "System.Data",
+    TypeName = "System.Data.Common.DbCommand",
+    MinimumVersion = "4.0.0",
+    MaximumVersion = "4.*.*",
+    IntegrationName = nameof(IntegrationId.AdoNet),
+    DataReaderType = TypeNames.DbDataReaderType,
+    DataReaderTaskType = TypeNames.DbDataReaderTaskType,
+    SignatureAttributes = new[]
+    {
+        // Task<int> System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken)
+        typeof(CommandExecuteNonQueryAsyncAttribute),
+        // Task<DbDataReader> System.Data.Common.DbCommand.ExecuteDbDataReaderAsync(CommandBehavior, CancellationToken)
+        typeof(CommandExecuteDbDataReaderWithBehaviorAndCancellationAsyncAttribute),
+        // Task<object> System.Data.Common.DbCommand.ExecuteScalarAsync(CancellationToken)
+        typeof(CommandExecuteScalarAsyncAttribute),
+    })]
 
-// Task<int> System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken)
-[assembly: CommandExecuteNonQueryAsync(typeof(SystemDataClientData))]
-[assembly: CommandExecuteNonQueryAsync(typeof(SystemDataCommonClientData))]
+[assembly: AdoNetClientInstrumentMethod(
```

Comment:
If it's not too much of a burden, if the rest of the team is on board with this PR, could you write a small README to document the ways that we specify integrations? It seems that there are now two well-defined ways:
1. At the type level, `InstrumentMethodAttribute`
2. At the assembly level, `AdoNetClientInstrumentMethodAttribute`

Reply:
Working on some docs 👍

Reply:
3. Added dynamically at run time calling `NativeMethods.InitializeProfiler()` directly (which #2233 is using)